### PR TITLE
[Backport 5.1]: grpc: moving IsGRPCEnabled to resolve import cycle issue (#55362)

### PR DESCRIPTION
**Pull request by mucles**
Moving IsGRPCEnabled from internal/grpc to internal/conf package to resolve the import cycle issue caused when using IsGRPCEnabled in the /internal/api/internalapi package.

This issues occurs because internalapi depends on -> internal/grpc (IsGRPCEnabled) depends on -> internal/conf -> which depends on internalapi

(cherry picked from commit 5831097001bb0fa97b2b8a6d26ed428c46bbbc04)

## Test plan
importing grpc/defaults to internal/api/internalapi/client.go and no longer got import cycle errors

### CODE REVIEW

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -1601,63 +1605,64 @@ func (c *clientImplementor) do(ctx context.Context, repoForTracing api.RepoName,
...
-			return nil, err
-		}
-		resp, err := client.CreateCommitFromPatchBinary(ctx, req.ToProto())
-		if err != nil {
-			return nil, err
-		}
+	if conf.IsGRPCEnabled(ctx) {
+		client, err := c.ClientForRepo(ctx, req.Repo)
+			if err != nil {
+				return nil, err
```
Comment (camdencheek) [inline internal/gitserver/client.go:1611]: What's with all the spacing changes in this file?

File: internal/gitserver/client.go [hunk 2]
```diff
@@ -699,11 +703,7 @@ func convertGRPCErrorToGitDomainError(err error) error {
 		return context.Canceled
 	}
 
-	if st.Code() == codes.DeadlineExceeded {
-		return context.DeadlineExceeded
-	}
-
-	for _, detail := range st.Details() {
```
Comment (camdencheek) [inline internal/gitserver/client.go:706]: Why was this removed?

File: internal/gitserver/client.go (hunk 1 above)
Comment (mucles) [reply internal/gitserver/client.go:1611]: @camdencheek  oops I was resolving the merge conflicts through vim that didn't work out so well 😆 , il fix it right now

File: internal/gitserver/client.go (hunk 2 above)
Comment (mucles) [reply internal/gitserver/client.go:706]: accidentally deleted it, adding it back.

**Review by camdencheek — APPROVED** [review]
